generators/app/index.js

```diff
@@ -222,53 +230,211 @@ module.exports = class extends Generator {
+                    return Promise.resolve();// {{ADS EDIT}}
                 }
 
-                generator.extensionConfig.isCustomization = true;
+                generator.extensionConfig.isCustomization = true;// {{ADS EDIT}}
 
-                return generator.prompt({
-                    type: 'confirm',
-                    name: 'addDashboardExtension',
-                    message: 'Add a full dashboard tab?',
-                    default: true
-                }).then(function (answer) {
-                    generator.extensionConfig.addDashboardTab = answer.addDashboardExtension;
-                    generator.extensionConfig.insightName = generator.extensionConfig.name + '.insight';
-                    generator.extensionConfig.tabName = generator.extensionConfig.name + '.tab';
+                return generator.prompt({// {{ADS EDIT}}
+                    type: 'confirm',// {{ADS EDIT}}
+                    name: 'addHomepageAction',// {{ADS EDIT}}
+                    message: 'Add a homepage action?',// {{ADS EDIT}}
+                    default: true// {{ADS EDIT}}
+                }).then(function (answer) {// {{ADS EDIT}}
+                    generator.extensionConfig.addHomepageAction = answer.addHomepageAction;// {{ADS EDIT}}
+                });
+            },
+
+            askForDashboardTab: () => {// {{ADS EDIT}}
+                if (generator.extensionConfig.type !== 'ext-dashboard') {// {{ADS EDIT}}
+                    return Promise.resolve();// {{ADS EDIT}}
+                }
+
+                generator.extensionConfig.isCustomization = true;// {{ADS EDIT}}
+
+                return generator.prompt({// {{ADS EDIT}}
+                    type: 'confirm',// {{ADS EDIT}}
+                    name: 'addDashboardExtension',// {{ADS EDIT}}
+                    message: 'Add a full dashboard tab?',// {{ADS EDIT}}
+                    default: true// {{ADS EDIT}}
+                }).then(function (answer) {// {{ADS EDIT}}
+                    generator.extensionConfig.addDashboardTab = answer.addDashboardExtension;// {{ADS EDIT}}
+                    generator.extensionConfig.insightName = generator.extensionConfig.name + '.insight';// {{ADS EDIT}}
+                    generator.extensionConfig.tabName = generator.extensionConfig.name + '.tab';// {{ADS EDIT}}
+                    if (!generator.extensionConfig.addDashboardTab){
+                        generator.extensionConfig.addDashboardBar = false;
+                    }
+                });
+            },
+
+            askForServerTab: () => {// {{ADS EDIT}}
+                if (generator.extensionConfig.type !== 'ext-dashboard' || !generator.extensionConfig.addDashboardTab) {// {{ADS EDIT}}
+                    return Promise.resolve();// {{ADS EDIT}}
+                }
+
+                generator.extensionConfig.isCustomization = true;// {{ADS EDIT}}
+
+                return generator.prompt({// {{ADS EDIT}}
```

Comment:
you don't need to put ADS Edit in every line. you can put it on top of a block or around something. if it's one line of inside a big VS code block then maybe put it in line